scripts/common.py

```diff
@@ -38,9 +39,34 @@
 NODE_MODULES_PATH = os.path.join(CURR_DIR, 'node_modules')
 FRONTEND_DIR = os.path.join(CURR_DIR, 'core', 'templates', 'dev', 'head')
 YARN_PATH = os.path.join(OPPIA_TOOLS_DIR, 'yarn-v1.17.3')
+OS_NAME = platform.system()
+ARCHITECTURE = platform.machine()
+
 # Add path for node which is required by the node_modules.
 os.environ['PATH'] = (
-    '%s/bin:' % NODE_PATH + '%s/bin:' % YARN_PATH + os.environ['PATH'])
+    '%s%sbin%s' % (NODE_PATH, os.sep, os.pathsep) +
+    '%s%s' % (NODE_PATH, os.pathsep) +
+    '%s%sbin%s' % (YARN_PATH, os.sep, os.pathsep) + os.environ['PATH'])
```

Comment:
Let's set `sys.path` instead, this feels very dangerous to me.

Reply:
But `sys.path` is not the same as the environment variable. 
```
In [1]: import sys

In [2]: import os

In [3]: sys.path
Out[3]:
['',
 '/home/cuichen/opensource/venv/bin',
 '/home/cuichen/opensource/venv/lib/python2.7',
 '/home/cuichen/opensource/venv/lib/python2.7/plat-x86_64-linux-gnu',
 '/home/cuichen/opensource/venv/lib/python2.7/lib-tk',
 '/home/cuichen/opensource/venv/lib/python2.7/lib-old',
 '/home/cuichen/opensource/venv/lib/python2.7/lib-dynload',
 '/usr/lib/python2.7',
 '/usr/lib/python2.7/plat-x86_64-linux-gnu',
 '/usr/lib/python2.7/lib-tk',
 '/home/cuichen/opensource/venv/local/lib/python2.7/site-packages',
 '/home/cuichen/opensource/venv/lib/python2.7/site-packages',
 '/home/cuichen/opensource/venv/local/lib/python2.7/site-packages/IPython/extensions',
 '/home/cuichen/.ipython']

In [4]: os.environ['PATH']
Out[4]: '/home/cuichen/opensource/venv/bin:/home/cuichen/bin:/usr/local/bin:/usr/local/sbin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/cuichen/.local/bin:/home/cuichen/.antigen/bundles/robbyrussell/oh-my-zsh/lib:/home/cuichen/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/git:/home/cuichen/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/pip:/home/cuichen/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/python:/home/cuichen/.antigen/bundles/zsh-users/zsh-syntax-highlighting:/home/cuichen/.antigen/bundles/zsh-users/zsh-autosuggestions'        
```